### ISSUE TITLE
Cleaner Translation API

### ISSUE THREAD

**Issue by dereuromark** [cakephp/cakephp#13142]
This is a (multiple allowed):

* [ ] bug
* [x] enhancement
* [x] feature-discussion (RFC)

* CakePHP Version: ~~4.x~~ 3.next

### What you did
```php
$this->Flash->success(__('The state machine item state history has been saved.'));
```
inside controller actions

### What happened
phpstan level 6/7
```
  63     Parameter #1 $message of method Cake\Controller\Component\FlashComponent::success() expects string, string|null given.  
  66     Parameter #1 $message of method Cake\Controller\Component\FlashComponent::error() expects string, string|null given.    
  90     Parameter #1 $message of method Cake\Controller\Component\FlashComponent::success() expects string, string|null given.  
  93     Parameter #1 $message of method Cake\Controller\Component\FlashComponent::error() expects string, string|null given.    
  114    Parameter #1 $message of method Cake\Controller\Component\FlashComponent::success() expects string, string|null given.  
  116    Parameter #1 $message of method Cake\Controller\Component\FlashComponent::error() expects string, string|null given.   
```

### What you expected to happen
I would like to use __() functions that for sure can not return null, they must return either a string (e.g. always empty string) or better yet throw exception in such cases.

**Comment by ADmad** [comment]
The error doesn't make sense, [return type](https://github.com/cakephp/cakephp/blob/4.x/src/I18n/functions.php#L28) for `__()` is typehinted as string.

**Comment by dereuromark** [comment]
Sry, I was running 3.x here. But if this cannot be fixed inside 3.x then I am ok with only having it in 4.0

**Comment by markstory** [comment]
This is already fixed in 4.0. I am not sure it is worth fixing for 3.8.
